CQC Rebalancing

Kell-E opened the pull request:
A few tweaks to the way that CQC works to reduce the absolutely absurd effect that the skill has on the outcome of everything that involves touching people.

To go into the exact changes, expect the following.
Accuracy penalties (And increases) per level difference have been decreased to 5% from 10%

Parry is set to the baseline of the weapon (15 / 30 / 50 in most cases) and then increases by 10 / 20% for experienced / master, down from 20% bonus parry per level of difference (This worked both ways, so one level of difference meant you gained a 20% chance to parry and a 20% decreased chance to be parried, which was absurd)

Parry is now capped at 75% down from being theoretically capable of hitting 140% or higher in some cases.

Disarm chances have been decreased and the effect that skills have on disarm has been reduced, there's now at least always a chance you can disarm / shove someone even if the chance gets extremely low in the absolute worst case scenario. 

Stimulants now actually have relevance in the disarm calculations

The chance to hurt your hand on your opponents weapon scales a little more with skill

Grab chances are more sane, the odds of being able to escape a grab will never hit 100% but they'll also scale down in much less dramatic fashion. In order to make a grab inescapable you'll need to have 3+ levels over your opponent, or 2+ and also have them disoriented via flash / pepper spray / flashbang / concussions / pain

I think that about covers everything, the grab chances definitely need to see some live playtesting before I am happy with the numbers but this should be a way more reasonable baseline. 

I genuinely think these changes are needed, the last time we saw master CQC traitors life became hellish for security and the same really does go both ways. Bringing the effectiveness of CQC down to a more reasonable level will allow for more freedom in skill choices going forwards, in order to participate in any kind of melee combat you'll only really need trained instead of being required to take at least experienced.

It's worth noting that the changes in this PR are reliant on the changes in #31035 in order to allow many of the common weapons to parry

:cl: Yvesza
balance: Reduces the effectiveness of CQC
tweak: You can no longer parry attacks with unsuitable weapons
/:cl:

SierraKomodo commented on code/game/objects/items.dm:
```diff
@@ -535,9 +533,10 @@ var/list/global/slot_flags_enumeration = list(
+	if (user.a_intent == I_HELP)
+		. = 0
+	if (.)
+		. += min(75, max(0, (user.get_skill_value(SKILL_COMBAT) * 10) - 20))
```
`Clamp()` does the same thing as `min(max())` without the double proc calls.

```suggestion
		. += Clamp((user.get_skill_value(SKILL_COMBAT) * 10) - 20, 0, 75)
```

Kell-E replied:
Got it! Thanks, I'll use that in future!

SierraKomodo commented on code/modules/mob/grab/grab_datum.dm:
```diff
@@ -300,6 +300,8 @@
 			affecting.visible_message("<span class='warning'>[affecting] has broken free of [assailant]'s grip!</span>")
 			let_go(G)
+	else
+		affecting.visible_message("<span class='warning'>[break_chance]</span>")
```
Use `SPAN_*` defines for these.

Also, this looks like it's just going to put a number into chat?

```suggestion
		affecting.visible_message(SPAN_WARNING("[break_chance]"))
```

Kell-E replied:
Whoops, I was just using that to check numbers were being used correctly. Missed taking it out!

SierraKomodo reviewed (changes requested):
I need to stop forgetting to check commit count.

Squash.

Kell-E committed:
Initial CQC Changes

Way more to come.
Initial CQC changes

Values subject to discussion and further changes, given how much of a balance impact this could have.
Quick fixes

Forgot to take out debug variables, whoops!
changes based upon review

Fixed a forgotten variable check, amended a min/max check to a clamp.

Kell-E commented:
Squashed! I still suspect that I'll need to do some fiddling with the grab values in the next week or so once I've seen it all in action, but I'll be keeping a -really- close eye on how it all plays out.
